api/v1/routes/waitlist.py

```diff
@@ -0,0 +1,35 @@
+from fastapi import APIRouter, HTTPException, Depends, Request
+from sqlalchemy.orm import Session
+from pydantic import BaseModel, EmailStr
+from api.v1.models.user import WaitlistUser
+from api.v1.schemas.waitlist import WaitlistUserCreate
+from api.db.database import get_db
+from api.v1.services.waitlist_email import send_confirmation_email
+
+router = APIRouter()
+
+
+class WaitlistResponse(BaseModel):
+    message: str
+
+
+@router.post("/waitlist", response_model=WaitlistResponse, status_code=201)
+@rate_limit(max_calls=5, time_frame=60)
+async def signup_waitlist(
+    request: Request,
+    user: WaitlistUserCreate,
```

Comment:
Did you confirm that this `rate_limit` decorator works as it should

Reply:
I tested it initially with the unit tests I wrote and it passed.

But with the email service not fully implemented yet, I couldn't re-test it.

Reply:
@Dev-wonderful I have fixed it now